langchain/src/document_loaders/web/wikipedia.ts

```diff
@@ -0,0 +1,131 @@
+import wiki, { Page } from "wikipedia";
+import { Document } from "../../document.js";
+import { BaseDocumentLoader } from "../base.js";
+
+const DISCARDCATEGORIES = [
+  "See also",
+  "References",
+  "External links",
+  "Further reading",
+  "Footnotes",
+  "Bibliography",
+  "Sources",
+  "Citations",
+  "Literature",
+  "Footnotes",
+  "Notes and references",
+  "Photo gallery",
+  "Works cited",
+  "Photos",
+  "Gallery",
+  "Notes",
+  "References and sources",
+  "References and notes",
+];
+
+export class WikipediaLoader extends BaseDocumentLoader {
+  discardCategories: string[];
+
+  constructor(public webPath: string, discardCategories?: string[]) {
+    super();
+    this.discardCategories = discardCategories ?? DISCARDCATEGORIES;
+  }
+
+  public async load(): Promise<Document[]> {
+    const matchResult = this.webPath.match(/\/wiki\/(.+)/);
+    if (!matchResult) {
+      throw new Error(
+        "Invalid webPath format. Use full url like https://en.wikipedia.org/wiki/2020_Summer_Olympics"
+      );
+    }
+
+    const pageTitle = decodeURIComponent(matchResult[1]);
```

Comment:
I think perform an actual search instead of decoding the last URL portion is more resilient and matches the behavior of Wikipedia loader.